opensearch_py_ml/ml_commons/ml_commons_client.py

```diff
@@ -138,53 +236,94 @@ def _send_model_info(self, model_meta_json: dict):
             if status["state"] != "CREATED":
                 task_flag = True
         if not task_flag:
-            raise TimeoutError("Uploading model timed out")
+            raise TimeoutError("Model registration timed out")
         if status["state"] == "FAILED":
             raise Exception(status["error"])
-        print("Model was uploaded successfully. Model Id: ", status["model_id"])
+        print("Model was registered successfully. Model Id: ", status["model_id"])
         return status["model_id"]
 
+    @deprecated(
+        reason="Since OpenSearch 2.7.0, you can use deploy_model instead",
+        version="2.7.0",
+    )
     def load_model(self, model_id: str, wait_until_loaded: bool = True) -> object:
         """
-        This method loads model into opensearch cluster using ml-common plugin's load model api
+        This method deploys a model in the opensearch cluster using ml-common plugin's deploy model api
 
         :param model_id: unique id of the model
         :type model_id: string
-        :param wait_until_loaded: Whether to wait until the model is loaded into memory
+        :param wait_until_loaded: Whether to wait until the model is deployed
         :type wait_until_loaded: bool
         :return: returns a json object, with task_id and status key.
         :rtype: object
         """
 
-        API_URL = f"{ML_BASE_URI}/models/{model_id}/_load"
+        API_URL = f"{ML_BASE_URI}/models/{model_id}/_deploy"
 
         task_id = self._client.transport.perform_request(method="POST", url=API_URL)[
             "task_id"
         ]
 
         if wait_until_loaded:
-            # Wait until loaded
+            # Wait until deployed
             for i in range(TIMEOUT):
                 ml_model_status = self.get_model_info(model_id)
                 model_state = ml_model_status.get("model_state")
-                if model_state in ["LOADED", "PARTIALLY_LOADED"]:
+                if model_state in ["DEPLOYED", "PARTIALLY_DEPLOYED"]:
                     break
                 time.sleep(1)
 
             # Check the model status
-            if model_state == "LOADED":
-                print("Model loaded into memory successfully")
-            elif model_state == "PARTIALLY_LOADED":
-                print("Model was loaded into memory only partially")
+            if model_state == "DEPLOYED":
+                print("Model deployed successfully")
+            elif model_state == "PARTIALLY_DEPLOYED":
+                print("Model deployed only partially")
             else:
-                raise Exception("Model load failed")
+                raise Exception("Model deployment failed")
```

Comment:
I think this can be covered in test, right?

Reply:
I'm not sure how I can force the deployment to fail